tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+        self.date = date
+        self.task = task
+        self.source_line = source_line
+
+    def __repr__(self):
+        return self.source_line
+
+    def __dict__(self):
+        return {
+            'type': self.type,
+            'date': self.date,
+            'time': self.time,
+            'verb': self.verb,
+            'task': self.task,
+            'extra': self.extra
+            }
+
+
+class LogReader:
+    """
+    LogReader manages the spread log, it allows to read, export and print a spread log.
+    """
+    def __init__(self, filepath, output_type, lines_limit, store_setup):
+        self.filepath = filepath
+        self.output_type = output_type
+        self.lines_limit = lines_limit
+        self.store_setup = store_setup
+        self.lines = []
+        self.iter = 0
+        self.full_log = []
+
+    def __repr__(self):
+        return str(self.__dict__())
+
+    def __dict__(self):
+        return {'full_log': self.full_log}
+
+    def print_log(self, details, results):
+        if not self.full_log:
+            return
+        if details == ALL:
+            print(''.join(str(x) for x in self.full_log))
+        elif details == NONE:
+            pass
+        elif details == ERROR:
+            print(''.join(str(x) for x in self.full_log if x.type == INFO and \
+                x.info_type == ERROR_TYPE))
+        else:
+            print(''.join(str(x) for x in self.full_log if x.type == details))
+
+        if results == ALL:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT))
+        elif results == NONE:
+            pass
+        elif results == FAILED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == FAILED_TYPE))
+        elif results == ABORTED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == ABORTED_TYPE))
+        else:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == SUCCESSFUL_TYPE))
+
+
+    def export_log(self, filepath):
+        prepared_log = []
+        for item in self.full_log:
+            prepared_log.append(item.__dict__())
+        with open(filepath, 'w') as json_file:
+            json.dump(prepared_log, json_file)
+
+    def _next_line(self):
+        self.iter = self.iter + 1
+        return self.lines[self.iter-1]
+
+    def check_log_exists(self):
+        return os.path.exists(self.filepath)
+
+    def read_spread_log(self):
+        with open(self.filepath) as filepath:
+            self.lines = filepath.readlines()
+            self.iter = 0
+
+            if self.store_setup:
+                while self.iter < len(self.lines):
+                    line = self._next_line()
+                    if self._match_start(line):
+                        break
+                    self.full_log.append(line)
+
+                if self.iter >= len(self.lines):
```

Comment:
If you use a normal `for` loop to iterate over `iter(self.lines)` the this can be the `else:` part of a `for` loop, which only executes if the loop does not invoke `break`.